core/Common.php

```diff
@@ -966,7 +966,7 @@ public static function extractLanguageAndRegionCodeFromBrowserLanguage($browserL
     {
         $validLanguages = self::checkValidLanguagesIsSet($validLanguages);
 
-        if (!preg_match_all('/(?:^|,)([a-z]{2,3})([-][a-z]{2})?/', $browserLanguage, $matches, PREG_SET_ORDER)) {
+        if (!preg_match_all('/(?:^|,)([a-z]{2,3})(?:[-][a-z]{4})?([-][a-z]{2})?/', $browserLanguage, $matches, PREG_SET_ORDER)) {
```

Comment:
not sure if I'm testing it correctly but below string didn't work for me where it maybe should have detect `zh-cn` but didn't

<img width="1069" alt="image" src="https://user-images.githubusercontent.com/273120/219787620-a3bb5377-9063-4d7f-80fc-4cc24fb808d4.png">

Reply:
actually it works with `zh-Hans-CN,zh;q=0.9`. I copied it wrongly from the PR description

Reply:
We are using `Common::getBrowserLanguage()` to get the browser language in most cases. That method already replaces `_` with `-` and lowers the string.